jans-cedarling/scripts/parse_binding_benchmarks.py

```
@@ -0,0 +1,180 @@
+    "wasm": ("WASM", parse_kv),
+    "java": ("Java", parse_kv),
+}
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("--format", required=True, choices=PARSERS, help="Benchmark output format")
+    parser.add_argument("file", nargs="?", help="Input file (default: stdin)")
+    args = parser.parse_args()
+
+    text = open(args.file).read() if args.file else sys.stdin.read()
```

Ruff (0.15.12)

[warning] Use a context manager for opening files

(SIM115)